tests/baselines/reference/emitter.asyncGenerators.functionExpressions.es2018.types

```diff
@@ -24,8 +24,8 @@ const f3 = async function * () {
 }
 === tests/cases/conformance/emitter/es2018/asyncGenerators/F4.ts ===
 const f4 = async function * () {
->f4 : () => AsyncIterableIterator<number>
->async function * () {    const x = yield* [1];} : () => AsyncIterableIterator<number>
+>f4 : () => AsyncGenerator<number, void, undefined>
+>async function * () {    const x = yield* [1];} : () => AsyncGenerator<number, void, undefined>

```

Comment:
Explained it [here](https://github.com/microsoft/TypeScript/pull/30790#discussion_r299222752) - `yield*` used `TReturn` and not `TYield`.

Reply:
What @sandersn is pointing to is the return type is `AsyncGenerator<number, void, undefined>`. This is because `yield* [1]` delegates to the iterator for `[1]`, which is `Iterator<number, void, undefined>`, so we *know* that the generator accepts `undefined` for `TNext` rather than `unknown`.